src/lib/y2storage/filesystems/btrfs.rb

```diff
@@ -80,6 +80,36 @@ def supports_btrfs_subvolumes?
         true
       end
 
+      # Check if a subvolume can be created.
+      #
+      # It can always be created if we're going to create the whole file
+      # system anyway.
+      #
+      # If the file system exists already there must at least be nothing
+      # else below path.
+      #
+      # @param path [String] subvolume path
+      #
+      # @return [Boolean]
+      #
+      def subvolume_can_be_created?(path)
+        return true unless exists_in_probed?
+        !subvolume_descendants_exist?(path)
+      end
+
+      # List of subvolumes hierarchically below path
+      #
+      # That is, subvolumes starting with path.
+      #
+      # @param path [String] subvolume path
+      #
+      # @return [Array<BtrfsSubvolume>]
+      #
+      def subvolume_descendants(path)
+        path += "/"
```

Comment:
I know we never use paths like "/var/" (trailing slash) internally. But I fear something like that can end up here via autoyast profile or lack of validation in the Partitioner. So better be safe with something like
```ruby
path += "/" unless path.end_with?("/")
```

Reply:
If that's the case the path should be put into a canonical form elsewhere. (I actually expected this.) Otherwise you'd have to deal with strange forms like `/foo///bar/` or whatever all over the place. At the very least provide a method for it.

BTW, just checked, and the path above is acceptable to the partitioner and it fails miserably when committing the change.